src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php

```diff
@@ -611,6 +612,49 @@ private static function analyzeOperands(
             return null;
         }
 
+        if ($parent instanceof PhpParser\Node\Expr\BinaryOp\Plus
+            || $parent instanceof PhpParser\Node\Expr\BinaryOp\Minus
+            || $parent instanceof PhpParser\Node\Expr\BinaryOp\Mul
+            || $parent instanceof PhpParser\Node\Expr\BinaryOp\Div
+            || $parent instanceof PhpParser\Node\Expr\BinaryOp\Mod
+            || $parent instanceof PhpParser\Node\Expr\BinaryOp\Pow
+        ) {
+            $non_decimal_type = null;
+            if ($left_type_part instanceof TNamedObject
+                && strtolower($left_type_part->value) === "decimal\\decimal"
+            ) {
+                $non_decimal_type = $right_type_part;
+            } elseif ($right_type_part instanceof TNamedObject
+                && strtolower($right_type_part->value) === "decimal\\decimal"
+            ) {
+                $non_decimal_type = $left_type_part;
+            }
+            if ($non_decimal_type !== null) {
+                if ($non_decimal_type instanceof TInt
+                    || $non_decimal_type instanceof TNumericString
+                    || $non_decimal_type instanceof TNamedObject
+                        && strtolower($non_decimal_type->value) === "decimal\\decimal"
+                ) {
+                    $result_type = Type::combineUnionTypes(
+                        new Type\Union([new TNamedObject("Decimal\\Decimal")]),
+                        $result_type
+                    );
+                } else {
+                    if ($statements_source && IssueBuffer::accepts(
+                        new InvalidOperand(
+                            "Cannot add Decimal\\Decimal to {$non_decimal_type->getId()}",
+                            new CodeLocation($statements_source, $parent)
+                        ),
+                        $statements_source->getSuppressedIssues()
+                    )) {
+                        // fall through
+                    }
```

Comment:
There's a new method called `IssueBuffer::maybyAdd()` with the same parameters, but `void` return. It makes sense to use it here instead of `IssueBuffer::accepts()`.